eng/tools/versioning/increment.js

```diff
@@ -0,0 +1,74 @@
+let argv = require("yargs")
+  .options({
+    "artifact-name": {
+      type: "string",
+      describe:
+        "name of the artifact to be incremented (e.g. azure-keyvault-secrets), will be translated to @azure/(package) format",
+      demandOption: true
+    },
+    "repo-root": {
+      type: "string",
+      describe: "root of the repository (e.g. ../../../)",
+      demandOption: true
+    },
+    "dry-run": {
+      type: "boolean"
+    }
+  })
+  .help().argv;
+
+const path = require("path");
+const semver = require("semver");
+const versionUtils = require("./VersionUtils");
+
+function incrementVersion(currentVersion) {
+  const prerelease = semver.prerelease(currentVersion);
+  if (prerelease) {
+    return semver.inc(currentVersion, "prerelease");
+  }
+
```

Comment:
At some point we might want to consider an option for incrementing the patch version for hotfixes.

Reply:
Being able to do this automatically with a `--hotfix` switch to the incrementer will improve automation. One can do this today by explicitly specifying the version in `set-version.js`... but having the automation is still better. I've created an issue to track this: #5911